server/rest/src/main/java/org/infinispan/rest/embedded/netty4/ChunkOutputStream.java

```diff
@@ -0,0 +1,92 @@
+package org.infinispan.rest.embedded.netty4;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import org.infinispan.rest.embedded.netty4.i18n.Messages;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpContent;
+
+/**
+ * Class to help application that are built to write to an
+ * OutputStream to chunk the content
+ * <p>
+ * <pre>
+ * {@code
+ * DefaultHttpResponse response = new DefaultHttpResponse(HTTP_1_1, OK);
+ * HttpHeaders.setTransferEncodingChunked(response);
+ * response.headers().set(CONTENT_TYPE, "application/octet-stream");
+ * //other headers
+ * ctx.write(response);
+ * // code of the application that use the ChunkOutputStream
+ * // Don't forget to close the ChunkOutputStream after use!
+ * ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT).addListener(ChannelFutureListener.CLOSE);
+ * }
+ * </pre>
+ *
+ * @author tbussier
```

Comment:
Should add what version this was copied from in case if we have to update/rebase on it in the future.  Same with the others.